### PR TITLE
Feature/simplify

### PR DESCRIPTION
**Pull request by Jammy2211**
Simplify the project a lot.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -310,140 +291,139 @@ jobs:
...
+#        git config --global user.name "GitHub Actions bot"
+#    - name: Update version
+#      run: |
+#        pushd "${{ matrix.project.path }}"
+#        VERSION="${{ needs.version_number.outputs.version_number }}"
+#        sed -i "s/__version__ = [\.\"\'0-9]*/__version__ = \"$VERSION\"/g" */__init__.py
+#        git commit "-am 'Updated version in __init__ to $VERSION"
+#    - name: Tag
+#      run: |
+#        pushd "${{ matrix.project.path }}"
```
Comment (rhayes777) [inline .github/workflows/release.yml:354]: I assume you want to coment this back in?

File: autobuild/build_util.py [hunk 2]
```diff
@@ -88,6 +76,30 @@ def execute_notebook(f):
         # subprocess.run(['jupyter', 'nbconvert', '--to', 'notebook', '--execute', f'{f}'], check=True)
```
Comment (rhayes777) [inline autobuild/build_util.py]: Do we want to remove this commented line?

File: autobuild/build_util.py [hunk 3]
```diff
@@ -88,6 +76,30 @@ def execute_notebook(f):
...
 
 
+def exexcute_notebooks_in_folder(ROOT_PATH, no_run_list=None):
```
Comment (rhayes777) [inline autobuild/build_util.py]: I keep meaning to fix this spelling

Comment (rhayes777) [reply autobuild/build_util.py]: In this context we could use a default value of () for no_run_list as tuples are immutable

File: autobuild/build_util.py [hunk 4]
```diff
@@ -88,6 +76,30 @@ def execute_notebook(f):
...
+
+        notebooks_path = f"{ROOT_PATH}/{x}"
+        os.chdir(notebooks_path)
+
+        for f in glob.glob(f"*.ipynb"):
+
+            run_notebook = True
+            for no_run in no_run_list:
+                if no_run in f:
+                    run_notebook = False
```
Comment (rhayes777) [inline autobuild/build_util.py]: There's no point defining this bool just invert the if statement and call execute_notebook(f)

File: autobuild/config/no_run.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+autogalaxy:
+- profiling
+- overview_10_clusters
+- tutorial_4_hierarchical_models
+- tutorial_5_expectation_propagation
+- Zeus
+- EmceePlotter
+- tutorial_searches
+- example_0
+- example_1
```
Comment (rhayes777) [inline autobuild/config/no_run.yaml:18]: oooo

File: autobuild/generate.py [hunk 6]
```diff
@@ -1,59 +1,57 @@
...
 
-import generate_util
 import build_util
+import generate_autofit
 
+project = sys.argv[1]
 
 BUILD_PATH = os.getcwd()
-WORKSPACE_PATH = f"{os.getcwd()}/../autofit_workspace"
+WORKSPACE_PATH = f"{os.getcwd()}/../{project}_workspace"
```
Comment (rhayes777) [inline autobuild/generate.py:14]: Prefer pathlib.Path

File: autobuild/generate.py [hunk 7]
```diff
@@ -1,59 +1,57 @@
...
-        if "__pycache__" in x:
-            try:
-                shutil.rmtree(x)
-            except FileNotFoundError:
-                pass
-
         scripts_path = f"{SCRIPTS_ROOT_PATH}/{x}"
         notebooks_path = f"{NOTEBOOKS_ROOT_PATH}/{x}"
-
+        if "__pycache__" in x:
```
Comment (rhayes777) [inline autobuild/generate.py]: I'm surprised there's __pycache__ directories present. That should only occur for local testing unless they got committed by mistake?

File: autobuild/generate.py [hunk 8]
```diff
@@ -64,7 +62,7 @@ def main():
         ### Copy notebooks to notebooks folder ###
 
         for f in glob.glob(f"*.ipynb"):
-            os.makedirs(notebooks_path, exist_ok=True)
+            os.makedirs(f"{notebooks_path}", exist_ok=True)
```
Comment (rhayes777) [inline autobuild/generate.py]: ? Did you want str(notebooks_path)?

File: autobuild/run.py [hunk 9]
```diff
@@ -0,0 +1,30 @@
+import os
+from os import path
+import sys
+import yaml
+
+import build_util
+
+os.environ["PYAUTOFIT_TEST_MODE"] = "1"
+
+project = sys.argv[1]
```
Comment (rhayes777) [inline autobuild/run.py:10]: We could also make these environment variables

**Review by rhayes777 — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (Jammy2211) [reply .github/workflows/release.yml:354]: Yep, once I've tested everything end-to-end.

**Pull request merged**